packages/block-editor/src/components/writing-flow/use-tab-nav.js

```diff
@@ -115,16 +104,22 @@ export default function useTabNav() {
 			// these are not rendered in the content and perhaps in the
 			// future they can be rendered in an iframe or shadow DOM.
 			if (
-				( isFormElement( event.target ) ||
+				( isFormElement( event.target, {
+					includeContentEditable: true,
+				} ) ||
 					event.target.getAttribute( 'data-block' ) ===
 						getSelectedBlockClientId() ) &&
-				isFormElement( focus.tabbable[ direction ]( event.target ) )
+				isFormElement( focus.tabbable[ direction ]( event.target ), {
+					includeContentEditable: true,
+				} ) &&
+				isInSameBlock(
```

Comment:
Because of the `isFormField()` taking in to account `contenteditable="true"`, I needed to ensure we would not leave the current block. So it compares with the current position `event.target` with the new position (next tabbable based on `event.target`.

Reply:
Why do we still need the other checks? Can't we just check `isInSameBlock`?